node-client/src/log.ts

```diff
@@ -0,0 +1,36 @@
+import querystring = require('querystring');
+import stream = require('stream');
+
+import { KubeConfig } from './config';
+import { WebSocketHandler } from './web-socket-handler';
+
+export class Log {
+    public 'handler': WebSocketHandler;
+
+    public constructor(config: KubeConfig) {
+        this.handler = new WebSocketHandler(config);
+    }
+
+    // TODO add support for limitBytes, previous, sinceSeconds and tailLines
+    public log(namespace: string, podName: string, containerName: string,
+                  follow: boolean,
+                  timestamps: boolean,
+                  stream: stream.Writable): Promise<void> {
+        const query = {
+            container: containerName,
+            follow: follow,
+            timestamps: timestamps,
+        };
+        const queryStr = querystring.stringify(query);
+        const path = `/api/v1/namespaces/${namespace}/pods/${podName}/log?${queryStr}`;
+        const promise = this.handler.connect(path, null, (streamNum: number, buff: Buffer) => {
+            const charBuff = Buffer.allocUnsafe(1)
+            charBuff.writeInt8(streamNum, 0)
+            stream.write(Buffer.concat([charBuff, buff]));
```

Comment:
Little bit ugly trick, but this allows to re-use the code from `WebSocketHandler` without having to it in a way that could break the other consumers of it (`Attach`, `Exec`).

A single `write()` call is issued on the stream because that tends to correspond to a line in the logs (although this is probably not guaranteed).

Reply:
Sorry can you explain why this is needed? This is probably a bug in `WebSocketHandler` that should be fixed instead of doing this here...

(and it's probably my bug :)